src/main.rs

```diff
@@ -46,126 +50,219 @@ impl LogLevel {
-    ] {
-        for entry in glob(path)?.flatten() {
-            log::info!("running required check {}", entry.to_string_lossy());
-            let output = Command::new("bash").arg("-C").arg(entry).output()?;
+    if !path_exists {
+        return Err(anyhow!("ParseError"));
+    }
+
+    for path in GREENBOOT_INSTALL_PATHS {
+        let gereenboot_wanted_path = format!("{}{}", path, "/check/wanted.d/*.sh");
+        for entry in glob(&gereenboot_wanted_path)?.flatten() {
+            log::info!("running wanted check {}", entry.to_string_lossy());
+            let output = Command::new("bash").arg("-C").arg(entry.clone()).output()?;
             if !output.status.success() {
                 // combine and print stderr/stdout
-                log::warn!("wanted script failed...");
+                log::warn!("wanted script {} failed!", entry.to_string_lossy());
             }
         }
     }
-    // if a command with restart option in systemd fails to start we don't get it as "failed"
-    // reversing the check makes sure that if by the time After=multi-user the service isn't running then it's failing at least
-    let output = Command::new("systemctl")
-        .arg("list-units")
-        .arg("--state")
-        .arg("active")
-        .arg("--no-page")
-        .arg("--output")
-        .arg("json")
-        .output()?;
-    let services: Vec<ServiceStatus> = serde_json::from_str(&String::from_utf8(output.stdout)?)?;
-    let ss: Vec<String> = services.iter().map(|x| x.unit.clone()).collect();
-    let active_units: HashSet<String> = HashSet::from_iter(ss);
-    for service in ["sshd.service", "NetworkManager.service"] {
-        if !active_units.contains(service) {
-            log::warn!("service {} failed, see journal", service);
-            failure = true;
-        }
+
+    if script_failure {
+        log::error!("Greenboot health-check failed!");
+        return Err(anyhow!("health-check failed!"));
     }
-    if failure {
-        for path in ["/etc/greenboot/red.d/*.sh"] {
-            for entry in glob(path)?.flatten() {
-                log::info!("running red check {}", entry.to_string_lossy());
-                let output = Command::new("bash").arg("-C").arg(entry).output()?;
-                if !output.status.success() {
-                    // combine and print stderr/stdout
-                    log::warn!("red script failed...");
-                }
+    Ok(())
+}
+
+fn run_red() -> Result<(), Error> {
+    for path in GREENBOOT_INSTALL_PATHS {
+        let red_path = format!("{}{}", path, "/red.d/*.*");
+        for entry in glob(&red_path)?.flatten() {
+            log::info!("running red check {}", entry.to_string_lossy());
+            let output = Command::new("bash").arg("-C").arg(entry.clone()).output()?;
+            if !output.status.success() {
+                // combine and print stderr/stdout
+                log::warn!("red script: {} failed!", entry.to_string_lossy());
```

Comment:
run_red() never returns error , but we expect it to fail below on line 160 run_red().expect("Red scripts failed!");

Reply:
Thank for pointing out, greenboot should with the execution process even if run_red() fails.